hadoop-ozone/integration-test-s3/src/test/java/org/apache/hadoop/ozone/s3/ProxyServer.java

```diff
@@ -0,0 +1,210 @@
+
+    server = new Server(proxyPort);
+    ServletContextHandler context = new ServletContextHandler();
+    context.setContextPath("/");
+
+    ProxyHandler proxyHandler = new ProxyHandler();
+    ServletHolder proxyHolder = new ServletHolder(proxyHandler);
+
+    proxyHolder.setInitParameter("proxyTo", "");
+
+
+    context.addServlet(proxyHolder, "/*");
+    server.setHandler(context);
+
+    LOG.info("ProxyServer initialized with endpoints: {}", s3gEndpoints);
+    LOG.info("Load balance strategy: {}", loadBalanceStrategy.getClass().getSimpleName());
+  }
+
+  public void start() throws Exception {
+    server.start();
+    LOG.info("Proxy started on http://{}:{}", host, port);
+  }
+
+  public void stop() throws Exception {
+    server.stop();
+    LOG.info("Proxy stopped on http://{}:{}", host, port);
+  }
+
+  /**
+   * ProxyHandler is a subclass of Jetty's ProxyServlet.Transparent.
+   * It implements logic for request rewriting, service handling,
+   * proxy response failure, and rewrite failure handling.
+   * This handler is mainly used to forward requests to different S3G endpoints
+   * based on the load balancing strategy, handle special HTTP headers (such as Expect),
+   * and manage exceptions during the proxy process.
+   */
+  public class ProxyHandler extends ProxyServlet.Transparent {
+
+    @Override
+    public void init() throws ServletException {
+      super.init();
+
+      LOG.info("MyProxyHandler initialized with {} endpoints",
+          s3gEndpoints != null ? s3gEndpoints.size() : 0);
+      if (s3gEndpoints != null) {
+        LOG.info("Endpoints: {}", s3gEndpoints);
+      }
+      LOG.info("Load balance strategy: {}", loadBalanceStrategy.getClass().getSimpleName());
+    }
+
+    @Override
+    protected String rewriteTarget(HttpServletRequest request) {
+
+      String baseUrl = loadBalanceStrategy.selectEndpoint(s3gEndpoints);
+
+      String requestUri = request.getRequestURI();
+      String queryString = request.getQueryString();
+
+      StringBuilder targetUrl = new StringBuilder(baseUrl);
+      if (requestUri != null) {
+        targetUrl.append(requestUri);
+      }
+      if (queryString != null) {
+        targetUrl.append('?').append(queryString);
+      }
+
+      String finalUrl = targetUrl.toString();
+      LOG.info("Rewriting target URL: [{}] {}", request.getMethod(), finalUrl);
+      return finalUrl;
+    }
+
+    @Override
+    protected void service(HttpServletRequest request, HttpServletResponse response)
+        throws ServletException, IOException {
+
+      // Remove the Expect header to avoid Jetty's 100-continue handling issue:
+      // In some scenarios (e.g. testPutObjectEmpty), when content-length != 0, Jetty triggers the 100-continue process,
+      // causing the server to wait for a 100 response code. However, Jetty only sends the 100 response
+      // when the InputStream is read, which can lead to request failures.
+      if (request.getHeader("Expect") != null) {
```

Comment:
Nit: Replace all the `Expect` with a constant (the http dependencies should have one).